src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java

```diff
@@ -139,6 +154,21 @@ public void saveKnownAllergy() {
 	@Then("the system adds known allergies into the allergies table")
 	public void systemAddsKnownAllergy() {
 		assertNotNull(addOrEditAllergyPage.getAllergiesList());
+	}
+
+	@And("a user clicks on the delete Allergy")
+	public void deleteAllergy() {
+		allergyPage.clickOnDeleteAllergy();
+	}
+
+	@And("a user confirms delete action")
+	public void confirmDeleteAllergy() {
+		allergyPage.clickOnConfirmDeleteAllergy();
+	}
+
+	@Then("the system deletes an allergy from the allergy table")
+	public void systemRemovesAllergy() {
+		assertTrue(textExists("Penicillins"));
```

Comment:
you can rather assert that the allergy table is not null after removing one of the allergy from the table